src/org/opensolaris/opengrok/history/BitKeeperRepository.java

```diff
@@ -0,0 +1,419 @@
+                working = Boolean.TRUE;
+                final Matcher matcher = VERSION_PATTERN.matcher(exec.getOutputString());
+                if (matcher.find()) {
+                    try {
+                        version = new Version(matcher.group(1));
+                    } catch (final InvalidVersionNumberException e) {
+                        assert false : "Failed to parse a version number.";
+                    }
+                }
+            } else {
+                working = Boolean.FALSE;
+            }
+            if (version == null) {
+                version = new Version(0, 0);
+            }
+        }
+    }
+
+    /**
+     * Returns whether file represents a BitKeeper repository. A BitKeeper repository has a folder named .bk at its
+     * source root.
+     *
+     * @return ret a boolean denoting whether it is or not
+     */
+    @Override
+    boolean isRepositoryFor(File file) {
+        if (file.isDirectory()) {
+            final File f = new File(file, ".bk");
+            return f.exists() && f.isDirectory();
+        }
+        return false;
+    }
+
+    /**
+     * Returns whether the BitKeeper command is working.
+     *
+     * @return working a boolean denoting whether it is or not
+     */
+    @Override
+    public boolean isWorking() {
+        ensureVersion();
+        return working.booleanValue();
+    }
+
+    /**
+     * Returns the version of the BitKeeper executable.
+     *
+     * @return version a Version object
+     */
+    public Version getVersion() {
+        ensureVersion();
+        return version;
+    }
+
+    /**
+     * Implementation of abstract method determineBranch. BitKeeper doesn't really have branches as such.
+     *
+     * @return null
+     */
+    @Override
+    String determineBranch() throws IOException {
+        return null;
+    }
+
+    /**
+     * Return the first listed pull parent of this repository BitKeeper can have multiple push parents and pul parents.
+     *
+     * @return parent a string denoting the parent, or null.
+     */
+    @Override
+    String determineParent() throws IOException {
+        final File directory = new File(directoryName);
+
+        final ArrayList<String> argv = new ArrayList<String>();
+        ensureCommand(CMD_PROPERTY_KEY, CMD_FALLBACK);
+        argv.add(RepoCommand);
+        argv.add("parent");
+        argv.add("-1il");
+
+        final Executor executor = new Executor(argv, directory);
+        final int rc = executor.exec(false);
+        final String parent = executor.getOutputString().trim();
+        if (rc == 0) {
+            return parent;
+        } else if (parent.equals("This repository has no pull parent.")) {
+            return null;
+        } else {
+            throw new IOException(executor.getErrorString());
+        }
+    }
+
+    /* History Stuff */
+    /*
+     * BitKeeper has independent revisions for its individual files like CVS, but also provides changesets, which is an
```

Comment:
can you break the lines to make them shorter? (there are more of those lines in the file)